File: demo-android/app/src/androidTest/java/com/example/demo/e2e/test/translation/LanguageSupportTest.kt
package com.example.demo.e2e.test.translation

import android.content.Context
import androidx.test.platform.app.InstrumentationRegistry
import com.example.demo.data.model.TaskPriority
import com.example.demo.data.model.TaskStatus
import com.example.demo.e2e.context.TaskTestContext
import com.example.demo.e2e.data.AllureEpic.TRANSLATION
import com.example.demo.e2e.data.TaskResponse
import com.example.demo.e2e.test.base.MockedBackendTestBase
import com.example.demo.locale.AppLocale
import io.qameta.allure.kotlin.Epic
import io.qameta.allure.kotlin.Feature
import io.qameta.allure.kotlin.TmsLink
import org.junit.Before
import org.junit.Test

@Epic(TRANSLATION)
@Feature("Language support")
@TmsLink("104")
class LanguageSupportTest : MockedBackendTestBase() {

    private val targetContext: Context
        get() = InstrumentationRegistry.getInstrumentation().targetContext

    private lateinit var testContext: TaskTestContext

    @Before
    fun setup() {
        testContext = TaskTestContext(
            status = TaskStatus.TODO,
            priority = TaskPriority.LOW,
        )

        support().mock.api()
            .getTasks(testContext.createTaskResponse().toTask())
            .getTasks(testContext.createTaskResponse().toTask())
    }

    @Before
    fun resetLocale() {
        targetContext.getSharedPreferences("demo_locale", Context.MODE_PRIVATE)
            .edit()
            .clear()
            .apply()
        AppLocale.setLanguage(targetContext, AppLocale.ENGLISH)
    }

    @Test
    fun shouldSwitchUiToSpanishWhenEsSelected() {
        // Given
        steps.navigation.openMainPage()

        // When
        steps.language.selectLanguage("ES")

        // Then
        validate.language.uiIsInSpanish()
        validate.language.statusTagShowsText(TaskStatus.TODO, "Por hacer")
        validate.language.priorityTagShowsText(TaskPriority.LOW, "Baja")
    }
}

File: demo-ios/DemoUITests/Test/Translation/LanguageSupportTest.swift
import XCTest

final class LanguageSupportTest: MockedBackendTestBase, AllureAnnotated {
    static var allureEpic: String { AllureEpic.translation }
    static var allureFeature: String { "Language support" }
    static var allureTmsLinks: [String] { ["104"] }

    private var testContext: TaskTestContext!

    override func stubApis() throws {
        testContext = TaskTestContext(status: .todo, priority: .low)
        let response = testContext.createTaskResponse()

        try support.mock.api()
            .getTasks(response)
            .getTasks(response)
    }

    func testShouldSwitchUiToSpanishWhenEsSelected() {
        // Given
        steps.navigation.openMainPage()

        // When
        steps.language.selectLanguage("ES")

        // Then
        validate.language.uiIsInSpanish()
        validate.language.statusTagShowsText(.todo, expectedText: "Por hacer")
        validate.language.priorityTagShowsText(.low, expectedText: "Baja")
    }
}
